### PR TITLE
Move console to console.ts and better stringify

### PR DESCRIPTION
**Pull request by kevinkassimo**
Moving console definition to own file `console.ts` and make it a class.
Replaced `JSON.stringify()` with custom `stringify()` to avoid crashing on circular objects and work with methods/functions by displaying `[Function]`.
Add some aliases to already implemented console methods.

### CODE REVIEW

File: console.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    } else {
+      out.push(stringify(new ConsoleContext(), a));
+    }
+  }
+  return out.join(" ");
+}
+
+const _print = V8Worker2.print;
+
+export class DConsole {
```
Comment (ry) [inline console.ts]: What does D stand for?

File: console.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    if (typeof a === "string") {
+      out.push(a);
+    } else {
+      out.push(stringify(new ConsoleContext(), a));
+    }
+  }
+  return out.join(" ");
+}
+
+const _print = V8Worker2.print;
```
Comment (ry) [inline console.ts]: The underscore is unnecessary.

File: console.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+export class DConsole {
+  // tslint:disable-next-line:no-any
+  log(...args: any[]): void {
+    _print(stringifyArgs(args));
+  }
+
+  debug = this.log;
+  info = this.log;
+  dirxml = this.log;
```
Comment (ry) [inline console.ts]: What's dirxml?

File: console.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
+class ConsoleContext {
```
Comment (ry) [inline console.ts]: why not
`type ConsoleContext = Set<any>;`

File: console.ts (hunk 1 above)
Comment (kevinkassimo) [reply console.ts]: I added it to suppress a warning of my editor. Should have removed it, will do right away

File: console.ts (hunk 2 above)
Comment (kevinkassimo) [reply console.ts]: Same as above (seems to be considered as print in browser)

File: console.ts (hunk 4 above)
Comment (kevinkassimo) [reply console.ts]: Not quite sure if there would be any more fields added to the ConsoleContext, so I left it that way so could later introduce new props if necessary. Could change to type though

**Comment by kevinkassimo** [comment]
Split the original commits to multiple, removed console.dirxml, changed ConsoleContext. Waiting for CI

**Comment by kevinkassimo** [comment]
Improved log format of functions, symbols, and class instances.

**Comment by kevinkassimo** [comment]
@ry Noticed that this PR has been held for quite a while. I can probably close it if it is not important for now

**Comment by ry** [comment]
@kevinkassimo Will get to it tomorrow - sorry for the delay!

**Comment by ry** [comment]
@kevinkassimo Thank you! It's a real improvement to the codebase. And apologies again for slow response. 
Tweaked slightly, squashed, rebased and landed in: 93654831fe501a4a011610bfc52b2d0fc87eb949

**Pull request closed without merging**